Doberman/AlarmMonitor.py

```diff
@@ -66,60 +66,57 @@ def send_phonecall(self, phone_numbers, message):
-                toaddr = toaddr.split(',')
-            recipients = toaddr
-            msg = MIMEMultipart()
-            msg['From'] = fromaddr
-            msg['To'] = ', '.join(toaddr)
-            if cc:
-                if not isinstance(cc, list):
-                    cc = cc.split(',')
-                msg['Cc'] = ', '.join(cc)
-                recipients.extend(cc)
-            if bcc:
-                if not isinstance(bcc, list):
-                    bcc = bcc.split(',')
-                msg['Bcc'] = ', '.join(bcc)
-                recipients.extend(bcc)
-            msg['Subject'] = subject
-            if add_signature:
-                signature = f'\n\n----------\nMessage created on {now} by Doberman slow control.'
-                body = str(message) + signature
-            else:
-                body = str(message)
-            msg.attach(MIMEText(body, 'plain'))
-
-            # Connect and send
-            if server_addr == 'localhost':  # From localhost
-                smtp = smtplib.SMTP(server_addr)
-                smtp.sendmail(fromaddr, toaddr, msg.as_string())
-            else:  # with e.g. gmail
-                server = smtplib.SMTP(server_addr, port)
-                server.starttls()
-                server.login(fromaddr, password)
-                server.sendmail(fromaddr, recipients, msg.as_string())
-                server.quit()
-            self.logger.info(f'Mail (Subject:{subject}) sent to {",".join(recipients)}')
-        except Exception as e:
-            self.logger.warning(f'Could not send mail: {e} ({type(e)})')
-            return 0
+            raise ValueError("No email connection details found")
+
+        # Compose connection details and addresses
+        now = dtnow().replace(tzinfo=timezone.utc).astimezone(tz=None).strftime("%Y-%m-%d %H:%M %Z")
+        server_addr = connection_details['server']
+        port = int(connection_details['port'])
+        fromaddr = connection_details['fromaddr']
+        password = connection_details['password']
+        if not isinstance(toaddr, list):
+            toaddr = toaddr.split(',')
+        recipients = toaddr
+        msg = MIMEMultipart()
+        msg['From'] = fromaddr
+        msg['To'] = ', '.join(toaddr)
+        if cc:
+            if not isinstance(cc, list):
+                cc = cc.split(',')
+            msg['Cc'] = ', '.join(cc)
+            recipients.extend(cc)
+        if bcc:
+            if not isinstance(bcc, list):
+                bcc = bcc.split(',')
+            msg['Bcc'] = ', '.join(bcc)
+            recipients.extend(bcc)
+        msg['Subject'] = subject
+        if add_signature:
+            signature = f'\n\n----------\nMessage created on {now} by Doberman slow control.'
+            body = str(message) + signature
+        else:
+            body = str(message)
+        msg.attach(MIMEText(body, 'plain'))
+
+        # Connect and send
+        if server_addr == 'localhost':  # From localhost
```

Comment:
Don't think so. I want all errors to be propagated to AlarmNode.log_alarm and discovered there rather than caught and ignored.

Reply:
Sure. I was just wondering what happens if `smtp = smtplib.SMTP(server_addr)` throws an [SMTPConnectError](https://docs.python.org/3/library/smtplib.html#smtplib.SMTPConnectError) or `smtp.sendmail(fromaddr, toaddr, msg.as_string())` an [SMTPRecipientsRefused](https://docs.python.org/3/library/smtplib.html#smtplib.SMTPRecipientsRefused) for example

Reply:
That would be the correct behaviour. This would throw the error which would be caught in the try except of line 65-74 of AlarmNode.py meaning that the pipeline would not be silenced. I think that's the same behaviour we had before, just by propagating the exception further rather than returning a 1